pypeman/contrib/http.py

```diff
@@ -112,7 +112,19 @@ def __init__(self, *args, endpoint=None, method='*', url='/', encoding=None, **k
 
     async def handle_request(self, request):
         content = await request.text()
-        msg = message.Message(content_type='http_request', payload=content, meta={'method': request.method})
+
+        # extract match info from iohttp request
+        meta = dict(request.match_info)
+
+        # match infos will be overwritten if coinciding with known keywords like
+        # 'url', 'method', . . .
+        meta.update({
+            'method': request.method,
+            'url': str(request.url),
```

Comment:
Why do you need `str` conversion here ?

Reply:
because request.url is a url object and not a string.
perhaps there's another method to convert to a string.

Somehow I thought it's better to have simple objects in the payload than url objects as it is easier to jsonify if needed.

Reply:
Ok fine !